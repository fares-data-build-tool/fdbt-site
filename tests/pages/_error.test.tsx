import * as React from 'react';
import { shallow } from 'enzyme';
import Error from '../../src/pages/_error';

describe('pages', () => {
    describe('operator', () => {
        it('should render error page correctly', () => {
            const tree = shallow(<Error error />);
            expect(tree).toMatchSnapshot();
        });

        it('should render 404 page correctly', () => {
            const tree = shallow(<Error />);
            expect(tree).toMatchSnapshot();
        });
    });
});
