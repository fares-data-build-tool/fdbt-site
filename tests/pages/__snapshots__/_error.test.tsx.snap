// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages operator should render 404 page correctly 1`] = `
<Layout
  description="Error page of the Fares data build tool"
  title="Error - Fares data build tool"
>
  <main
    className="govuk-main-wrapper app-main-class"
    id="main-content"
    role="main"
  >
    <div
      className="govuk-grid-row"
    >
      <div
        className="govuk-grid-column-two-thirds"
      >
        <div>
          <h1
            className="govuk-heading-xl"
          >
            Page not found
          </h1>
          <p
            className="govuk-body"
          >
            If you typed the web address, check it is correct.
          </p>
          <p
            className="govuk-body"
          >
            If you pasted the web address, check you copied the entire address.
          </p>
          <p
            className="govuk-body"
          >
            If the web address is correct or you selected a link or button,
             
            <a
              className="govuk-link"
              href="mailto:fdbt@transportforthenorth.com?bcc=tfn-feedback@infinityworks.com"
              id="feedback_link"
            >
              contact
            </a>
             
            us about your fares data.
          </p>
          <p
            className="govuk-body"
          >
            Click continue to start again.
          </p>
        </div>
        <br />
        <a
          className="govuk-button govuk-button--start"
          data-module="govuk-button"
          draggable="false"
          href="operator"
          role="button"
        >
          Continue
        </a>
      </div>
    </div>
  </main>
</Layout>
`;

exports[`pages operator should render error page correctly 1`] = `
<Layout
  description="Error page of the Fares data build tool"
  title="Error - Fares data build tool"
>
  <main
    className="govuk-main-wrapper app-main-class"
    id="main-content"
    role="main"
  >
    <div
      className="govuk-grid-row"
    >
      <div
        className="govuk-grid-column-two-thirds"
      >
        <div>
          <h1
            className="govuk-heading-xl"
          >
            Sorry, there is a problem with the service.
          </h1>
          <p
            className="govuk-body"
          >
            Try again later.
          </p>
          <p
            className="govuk-body"
          >
            Your answers have not been saved, click continue to start again.
          </p>
          <p
            className="govuk-body"
          >
            Contact us with the feedback button at the top of this page for assistance.
          </p>
        </div>
        <br />
        <a
          className="govuk-button govuk-button--start"
          data-module="govuk-button"
          draggable="false"
          href="operator"
          role="button"
        >
          Continue
        </a>
      </div>
    </div>
  </main>
</Layout>
`;
